### DISCUSSION TITLE
There is no data displayed in the data preview on ATTU, and data queries cannot retrieve data. How to delete entities

### DISCUSSION THREAD

**Discussion by opendatamarket** [milvus-io/milvus#25559] [Q&A and General discussion]
1、There is no data display in Data Preview on ATTU, and Data Query cannot retrieve the data
<img width="1146" alt="image" src="https://github.com/milvus-io/milvus/assets/106378749/dc2ad63a-fcac-43ca-9f02-7b97dcc0565d">
<img width="1172" alt="image" src="https://github.com/milvus-io/milvus/assets/106378749/3264793a-bedc-496e-bea6-ebb114c78ec1">
<img width="1154" alt="image" src="https://github.com/milvus-io/milvus/assets/106378749/a0242e89-8953-4e2a-bd47-6cafd823e9d9">

2、I need to implement a function, the name of the Collection is PDF_ Data, there is a role inside_ The ID field and PDF content are updated, so the role needs to be deleted_ I found that the document states that only the ID primary key can be used to delete the ID field, but I executed the following code without any effect
<img width="767" alt="image" src="https://github.com/milvus-io/milvus/assets/106378749/bf1d4d3b-6f4c-4815-8490-fad611e91d17">

3、How to obtain the ID of an entity? I would like to obtain a PDF_ Middle role of data_ What is the method for all entity IDs with id=10? I used the search API, but it prompted me to use vector search criteria, but I don't have vectors as criteria, only roles_ ID field

**Comment by yhmo** [comment]
1. Not sure what operations have done to the collection. Any deletion? The Attu version is 2.2.7. Which version of your Milvus server?

2. Your expression is "id in ["+id+"]", the "id" is the primary key, the deletion is done successfully. Always check the returned response:
R<MutationResult> response = milvusClient.delete(xxx);
assert (response.getStatus() == R.Status.Success.getCode())

3. Use SearchParam.withExpr() to set filtering expression. Read the Java SDK examples to learn more about how to call these interfaces: https://github.com/milvus-io/milvus-sdk-java/blob/2.2/examples/main/java/io/milvus/GeneralExample.java

Note: search() interface can return all kinds of field data except vector field data.

```
        SearchParam searchParam = SearchParam.newBuilder()
                .withCollectionName(COLLECTION_NAME)
                .withMetricType(MetricType.L2)
                .withOutFields(outFields)
                .withTopK(SEARCH_K)
                .withVectors(vectors)
                .withVectorFieldName(VECTOR_FIELD)
                .withExpr("id == 10")
                .withParams(SEARCH_PARAM)
                .withGuaranteeTimestamp(Constant.GUARANTEE_EVENTUALLY_TS)
                .build();

        R<SearchResults> response = milvusClient.search(searchParam);
```

**Comment by talhaanwarch** [comment]
i am also noticing this issue, as i re-install attu . installing 2.2.4 resolved the issue
